### ISSUE TITLE
NodeProxy: external server support broken

### ISSUE THREAD

**Issue by LFSaw** [supercollider/supercollider#3098]
This works

```
s.waitForBoot{
	n = NodeProxy(s, \audio, 2);
	n[0] = {SinOsc.ar};
	n.play;
}
```

This doesn't:

```
(
s = Server(\ext, NetAddr("localhost", 57110));
s.waitForBoot{
	n = NodeProxy(s, \audio, 2);
	n[0] = {SinOsc.ar};
	n.play;
}
)

// FAILURE IN SERVER /s_new duplicate node ID
```

**Comment by LFSaw** [comment]
I broke it down to NodeProxy... (and edited the description)

**Comment by LFSaw** [comment]
Let's see what should happen (as in what happens on `localhost`):


## localhost server


Preparation

```
(
s = Server.local;
s.waitForBoot{
	s.makeWindow;
	s.dumpOSC(1);
	CmdPeriod.add({s.freeAll});
};
)
// [ "/notify", 1 ]

// creating a NP does not send anything to the server (nice :)  )
n = NodeProxy(s, \audio, 2);
```

assigning a function creates a group and a synth (rendering the function) within

```
n[0] = {SinOsc.ar};
// -> NodeProxy.audio(ext, 2)
```

accordingly, the dumpOSC shows; interestingly, this is split into two budles (why? but the sync op prevents runtime errors, right?)

```
[ "#bundle", 1,
	[ 5, DATA[476] ],     // recv (different length of DATA[] than in \ext example, huh?
	[ 21, 1000, 1, 1 ],   // g_new 1000
	[ "/sync", 1002 ]     // sync
]
[ "#bundle", 15937675802586074463,

	                      // s_new 1001 in 1000
  [ 9, "temp__0940272221_0", 1001, 1, 1000, "out", 4, "i_out", 4 ]
]
```

now, play the NP, this should create two mono-Monitor synths within a new group

```
n.play;
// -> NodeProxy.audio(ext, 2)
```

I here added descriptions of the osc messages for your(my) convenience.


```
[ "#bundle", 15937676226404202247,
  [ 21, 1002, 1, 1 ],    // g_new 1002
                         // s_new 1003 in 1002
  [ 9, "system_link_audio_1", 1003, 1, 1002, "out", 0, "in", 4, "level", 1, "fadeTime", 0.02, "vol", 1 ],
                         // s_new 1004 in 1002
  [ 9, "system_link_audio_1", 1004, 1, 1002, "out", 1, "in", 5, "level", 1, "fadeTime", 0.02, "vol", 1 ]
]
```









```
s.queryAllNodes
```

results in

```
[ "/g_dumpTree", 0, 0 ]
NODE TREE Group 0
   1 group
      1000 group
         1001 temp__0940272221_0
      1002 group
         1003 system_link_audio_1
         1004 system_link_audio_1
```



------------------------------------------

## External server


Preparation

```
(
s = Server(\ext, NetAddr("localhost", 57110));
s.waitForBoot{
	s.makeWindow;
	s.dumpOSC(1);
	CmdPeriod.add({s.freeAll});
};
)
// [ "/notify", 1 ]

// creating a NP does not send anything to the server (nice :)  )
n = NodeProxy(s, \audio, 2);
```

assigning a function creates a group and a synth (rendering the function) within

```
n[0] = {SinOsc.ar};
// -> NodeProxy.audio(ext, 2)
```

accordingly, the dumpOSC shows; interestingly, this is split into two budles (why? but the sync op prevents runtime errors, right?)

```
[ "#bundle", 1,
	[ 5, DATA[478] ],     // recv
	[ 21, 1000, 1, 1 ],   // g_new 1000
	[ "/sync", 1002 ]     // sync
]
[ "#bundle", 15937672239760904490,

	                      // s_new 1001 in 1000
	[ 9, "temp__0-1789179580_0", 1001, 1, 1000, "out", 4, "i_out", 4 ]
]
```

now, play the NP, this should create two mono-Monitor synths within a new group

```
n.play;
// -> NodeProxy.audio(ext, 2)
```

I here added descriptions of the osc messages for your(my) convenience.
Apparently, the `system_link_audio_1` Synth is created twice, once with the nodeID of the already running synth

```
[ "#bundle", 15937672343625733151,
	[ 21, 1000, 1, 1 ],   // g_new 1000
                          // s_new 1001 in 1000
	[ 9, "system_link_audio_1", 1001, 1, 1000, "out", 0, "in", 4, "level", 1, "fadeTime", 0.02, "vol", 1 ],

	                      // s_new 1002 in 1000
	[ 9, "system_link_audio_1", 1002, 1, 1000, "out", 1, "in", 5, "level", 1, "fadeTime", 0.02, "vol", 1 ]
]
```

resulting in a

```
FAILURE IN SERVER /s_new duplicate node ID
```

```
s.queryAllNodes
```

results in

```
[ "/g_dumpTree", 0, 0 ]
NODE TREE Group 0
   1 group
      1000 group
         1001 temp__0-1789179580_0
         1002 system_link_audio_1
```

**Comment by LFSaw** [comment]
looks like that the NodeProxy uses the wrong `NodeAllocator` somewhere...

**Comment by LFSaw** [comment]
ok, I got it fixed with PR #3103 

Details:
```
(
s = Server(\ext, NetAddr("localhost", 57110));
s.waitForBoot{
	s.makeWindow;
	s.dumpOSC(1);
	CmdPeriod.add({s.freeAll});
};
)

n = NodeProxy(s, \audio, 2);

n.homeServer
n.monitor
n[0] = {SinOsc.ar};
n[1] = {SinOsc.ar};
n[2] = {SinOsc.ar};



n.play; // BusPlug.play
//	this.playToBundle(bundle, out, numChannels, group, multi, vol, fadeTime, addAction);
// group is nil but Monitor gets its information in Monitor:play via
// 			group.asGroup.server
// i.e.
nil.asGroup.server
// which results in \localhost (of course...)
// therefore
n.monitor.group.server // localhost
```
